src/dashboard/dashboard_manager.rs

```diff
@@ -0,0 +1,60 @@
+use std::io::Stdout;
+use std::{error::Error, io};
+
+use termion::raw::RawTerminal;
+use termion::{input::MouseTerminal, raw::IntoRawMode, screen::AlternateScreen};
+use tokio::task::LocalSet;
+use tui::{backend::TermionBackend, Terminal};
+
+use crate::client::globalsettings::GlobalSettings;
+use crate::common::WrappedRcRefCell;
+use crate::dashboard::dashboard_events::DashboardEventHandler;
+use crate::dashboard::dashboard_state::DashboardState;
+use crate::dashboard::ui::dashboard_ui::DashboardPainter;
+use crate::dashboard::ui::start::start_ui_loop;
+
+/// Allows drawing to the terminal
+pub type THandler = Terminal<TermionBackend<AlternateScreen<MouseTerminal<RawTerminal<Stdout>>>>>;
+
+/// The dashboard ui core structure
+pub struct DashboardManager {
+    pub dashboard_state: WrappedRcRefCell<DashboardState>,
+    global_settings: GlobalSettings,
+    terminal: WrappedRcRefCell<DashboardPainter>,
+    event_handler: DashboardEventHandler,
+}
+
+impl DashboardManager {
+    /// Initialises with the connection to the terminal and provides an event_handler to
+    /// update the UI
+    pub fn new(global_settings: GlobalSettings) -> Result<Self, io::Error> {
```

Comment:
Maybe the `DashboardManager` is useless for now? You can just pass these arguments directly to `start_ui_loop`.